cdap-security/src/main/java/co/cask/cdap/security/impersonation/RemoteUGIProvider.java

```diff
@@ -61,17 +63,20 @@
 
   @Inject
   RemoteUGIProvider(CConfiguration cConf, final DiscoveryServiceClient discoveryClient,
-                    LocationFactory locationFactory) {
-    super(cConf);
+                    LocationFactory locationFactory, OwnerAdmin ownerAdmin, NamespaceQueryAdmin namespaceQueryAdmin) {
+    super(cConf, ownerAdmin);
     this.remoteClient = new RemoteClient(discoveryClient, Constants.Service.APP_FABRIC_HTTP,
                                          new DefaultHttpRequestConfig(false), "/v1/");
     this.locationFactory = locationFactory;
   }
 
   @Override
   protected UGIWithPrincipal createUGI(ImpersonationRequest impersonationRequest) throws IOException {
+    ImpersonationRequest jsonRequest = new ImpersonationRequest(impersonationRequest.getEntityId(),
```

Comment:
Since the impersonation request in the param will include the key tab uri, and we do not want to pass that to master